source/common/http/conn_manager_impl.cc

```diff
@@ -708,6 +721,14 @@ void ConnectionManagerImpl::ActiveStream::onRequestTimeout() {
                  StreamInfo::ResponseCodeDetails::get().RequestOverallTimeout);
 }
 
+void ConnectionManagerImpl::ActiveStream::onRequestHeaderTimeout() {
+  connection_manager_.stats_.named_.downstream_rq_header_timeout_.inc();
+  sendLocalReply(request_headers_ != nullptr &&
```

Comment:
I recommend just closing the H1 connection on timeout.

We should think how to generalize this class of timeout to H2 and H3 cases.  We could apply the same timeout on the H2 connection when there are 0 active requests.

Reply:
There's already a connection idle timer that covers the 0 active request periods.

Reply:
Can we reset the stream in this case instead of generating a local reply?

Reply:
hm, I'd be mildly inclined to best-effort attempt to send a reply in case someone ratchets this up and it impacts folks with really crummy bandwidth.  WDYT?

Reply:
Sounds like sendLocalReply is preferred.  SGTM.